pkg/gds/gds.go

```diff
@@ -140,6 +136,19 @@ func (s *GDS) Register(ctx context.Context, in *api.RegisterRequest) (out *api.R
 		log.Warn().Err(err).Msg("ignoring validation error")
 	}
 
+	// Retrieve email address from one of the supplied contacts.
+	var email string
+	if email = getContactEmail(vasp); email == "" {
+		log.Warn().Msg("no contact email address found")
+		return nil, status.Error(codes.InvalidArgument, "no email address in supplied VASP contacts")
```

Comment:
In the [`vasp.Validate()`](https://github.com/trisacrypto/trisa/blob/main/pkg/trisa/gds/models/v1beta1/models.go#L107-L132) step we do check to make sure there is at least one contact and that all contacts supplied have email addresses. So these lines of code should never fire. Because of that could we change this to:

```suggestion
		log.Error().Err(errors.New("no contact email address found")).Msg("incorrect access on validated VASP")
		return nil, status.Error(codes.InvalidArgument, "no email address in supplied VASP contacts")
```

E.g. I want to make sure we get a critical alert if this case fires for some reason. 

As for the return - I'm happy to return an invalid argument, but only if we haven't saved anything to the database prior to this step. Would you mind checking to make sure there are no database modifications before we get here? (Otherwise we should return aborted and raise the stakes even higher for the log alert).